### PR TITLE
LPS-173591

### PR DESCRIPTION
**Pull request by joshuacords**
Hi @felipelorenz,

This change is good, we also need to write a test for this. Either directly reproduce the issue (permissioned folders, etc) or simply checking if what we search will take into account the Accurate Threshold Count.

### CODE REVIEW

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchIndexSearcher.java [hunk 1]
```diff
@@ -85,15 +85,19 @@ public Hits search(SearchContext searchContext, Query query) {
...
+
+			if (from == null) {
 				start = 0;
 			}
-			else if ((from != null) && (size != null) && (size > end)) {
-				end = from + size;
+			else {
+				end = from + end;
 				start = from;
 			}
```
Comment (gustavolimav) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchIndexSearcher.java:102]: Shouldn't it be?

	if ((size != null) && (size > end)) {
				if ((from == null)) {
					end = size;
					start = 0;
				}
				else {
					end = from + size;
					start = from;
				}
			}

Comment (gustavolimav) [reply modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchIndexSearcher.java:102]: @joshuacords

Comment (joshuacords) [reply modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchIndexSearcher.java:102]: @gustavolimav, I think you're right. But why exactly we don't ever run `end = from + end` makes me uncomfortable. In general the `end` variable is overloaded here. I think this is the simplest fix but I'm not sure it's the best.

**Comment by joshuacords** [comment]
We've taken a different approach.

**Pull request closed without merging**